rfcs/rdm-0064-custom-fields.md

```diff
@@ -0,0 +1,440 @@
+- I want to define in which order they will display inside those groups.
+- I want to add fields to already existing groups (e.g. Basic information).
+- I want to support translations (i18n) of the fields' label and description.
+
+**When displaying the custom fields of records...**
+
+- I want to define where and how the field will be shown in the landing/community page.
+- I want to define how the values will be serialized into different formats (e.g. Dublin Core, Darwin Core, DataCite).
+- I want to be able to filter my search results based on my new field.
+- I want to support translations (i18n) for the values of the custom fields.
+
+### Fist iteration limitations
+
+The first iteration of custom fields will be available in InvenioRDM v10. It targets the implementation of two vertical workflows, allowing to add free text custom fields (e.g. keywords, descriptions) and also based on generic vocabularies (e.g. defining a new "experiment" vocabulary). This will enable the implementation of the mechanisms to support custom fields, and allow developers to extend support for new types of fields (e.g. dates).
+
+Therefore, it **comes with some limitation**:
+
+- Only two field types are supported: Text and generic vocabularies. For example, resource types, or new ones of your creation that follow the `VocabularySchema`. This means, adhering to the following structure:
+
+```yaml
+id: ...
+icon: ...
+props: ...
+description: ...
+tags: ...
+title: ...
+```
+
+- Fields are configured globally for all records and for all communities. Note that they do not mix, those appearing in records will not appear in communities. However, all communities will have the same fields.
+- You can hide (not display) fields in the UI. However, they will be available in the REST API. Moreover, hiding them will hide them globally (i.e. for all records or communities).
+- There are no per-field permissions checks on these fields. They do respect the records permissions (i.e. will not show if the record is restricted).
+- It is not possible to configure the fields using the UI, it will be done in the instance `invenio.cfg` file.
+- Compatibility with OpenSearch has not been tested. Only Elasticsearch 7 has been tested.
+- In the forms, you can add fields to new groups but not to existing ones (e.g. Basic Information).
+- Translations (i18n) is only supported for labels and descriptions of all fields. Vocabularies also support translation of the values as long as those have been added when imported.
+
+## Detailed design
+
+An instance administrator can add custom fields. **Updating and deleting a custom field is not possible**. This constraint is imposed to avoid data migration and potential inconsistencies.
+
+### Configuring a _custom field_
+
+The first step would be to define how custom fields can be added/configured. A top to bottom approach (from more to less abstracted) will be followed to explain the configuration. Taking the following use case:
```

Comment:
I would use the present tense, as we are explaining the implementation.
```suggestion
The first step is to define how custom fields can be added/configured. A top to bottom approach (from more to less abstracted) will be followed to explain the configuration. Taking the following use case:
```

Reply:
fixed, also all other `would be` ocurrances.